src/StakeRegistry.sol

```diff
@@ -79,53 +79,42 @@ contract StakeRegistry is VoteWeigherBase, StakeRegistryStorage {
         // if they are, get their new weight and update their individual stake history and the
         // quorum's total stake history accordingly
         for (uint8 quorumNumber = 0; quorumNumber < quorumCount; ) {
-            OperatorStakeUpdate memory totalStakeUpdate;
-            // for each operator
+            int256 totalStakeDelta;
+
             for (uint i = 0; i < operators.length; ) {
                 bytes32 operatorId = registryCoordinator.getOperatorId(operators[i]);
                 uint192 quorumBitmap = registryCoordinator.getCurrentQuorumBitmapByOperatorId(operatorId);
-                // if the operator is not a part of any quorums, skip
-                if (quorumBitmap == 0) {
-                    continue;
-                }
-                // if the operator is a part of the quorum
+
+                /**
+                 * If the operator is a part of the quorum, update their current stake
+                 * and apply the delta to the total
+                 */ 
                 if (BitmapUtils.numberIsInBitmap(quorumBitmap, quorumNumber)) {
-                    // if the total stake has not been loaded yet, load it
-                    if (totalStakeUpdate.updateBlockNumber == 0) {
-                        totalStakeUpdate = _totalStakeHistory[quorumNumber][
-                            _totalStakeHistory[quorumNumber].length - 1
-                        ];
-                    }
-                    // update the operator's stake based on current state
-                    (uint96 stakeBeforeUpdate, uint96 stakeAfterUpdate) = _updateOperatorStake({
+                    (int256 stakeDelta, ) = _updateOperatorStake({
                         operator: operators[i],
                         operatorId: operatorId,
                         quorumNumber: quorumNumber
                     });
-                    // calculate the new total stake for the quorum
-                    totalStakeUpdate.stake = totalStakeUpdate.stake - stakeBeforeUpdate + stakeAfterUpdate;
+
+                    totalStakeDelta += stakeDelta;
                 }
                 unchecked {
                     ++i;
                 }
             }
-            // if the total stake for this quorum was updated, record it in storage
-            if (totalStakeUpdate.updateBlockNumber != 0) {
-                // update the total stake history for the quorum
-                _recordTotalStakeUpdate(quorumNumber, totalStakeUpdate);
+
+            // If we have a change in total stake for this quorum, update state
+            // TODO - do we want to record the update, even if the delta is zero?
+            //        maybe this makes sense in the case that the quorum doesn't have
+            //        an update for this block?
+            if (totalStakeDelta != 0) {
+                _recordTotalStakeUpdate(quorumNumber, totalStakeDelta);
```

Comment:
Question for reviewers - this one is similar to the other Q highlighted in this PR. Currently, the semantics here are a little weird - we basically push a stake update for a quorum if an operator belongs to the quorum, and we don't push anything if no operator belonged to that quorum. This also means that if 2 operators had a net-zero impact on the total, we'd push an update that ultimately didn't change the stake.

The current PR does things a little differently, and just checks each quorum to see if the delta is nonzero before pushing an update. This is similar to the current behavior in that if there were no operators belonging to the quorum, nothing gets pushed.

It differs because it also won't push if the delta was 0.

My question is - what do we want here?
- we can keep this behavior and just not push updates if there was no change
- or, we can change this slightly to also record net-zero updates if the last update is not the current block

Reply:
not pushing during net zero, should be fine :+1: